### PR TITLE
[Server] Lightning RPC + LevelDB + IslandPattern

### PR DESCRIPTION
**Pull request by mitchelljustin**
TODO: purchasing Hexalots

This includes results from pair programming on Thursday

### CODE REVIEW

File: server/src/hexalot.ts [hunk 1]
```diff
@@ -62,6 +62,29 @@ function* genSpiral(radius: number): IterableIterator<Vector> {
     }
 }
 
+export function overlap(h1: Hexalot, h2: Hexalot, direction: Direction): boolean {
```
Comment (mitchelljustin) [inline server/src/hexalot.ts:65]: Will probably be replaced by the same logic as you wrote client-side

File: server/index.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    const payments = await setupPaymentHandler()
+
+    const app = express()
+
+    app.use(bodyParser())
+
+    app.get("/test", (req, res) => res.end("OK"))
+
+    app.use("/island/genesis",
+        new IslandCurator("genesis", db, payments).createRouter())
```
Comment (mitchelljustin) [inline server/index.ts]: @geralddejong Got a better name for the "genesis" island?

Comment (geralddejong) [reply server/index.ts]: I expect we will have a number of islands.  Can it be "/island/{island-name}" ?  the client can decide which island to load.

File: server/src/hexalot.ts (hunk 1 above)
Comment (geralddejong) [reply server/src/hexalot.ts:65]: In a subsequent PR?

Comment (mitchelljustin) [reply server/src/hexalot.ts:65]: Yep

File: client/src/island/hexalot.ts [hunk 3]
```diff
@@ -17,7 +17,9 @@ interface IHexalotIndexed {
 
 const spotsToHexFingerprint = (spots: Spot[]): string => {
     const lit = spots.map(spot => spot.surface === Surface.Land ? "1" : "0")
-    const nybbleStrings = lit.map((l, index, array) => (index % 4 === 0) ? array.slice(index, index + 4).join("") : null).filter(chunk => chunk)
+    const nybbleStrings = lit
+        .map((l, index, array) => (index % 4 === 0) ? array.slice(index, index + 4).join("") : null)
```
Comment (geralddejong) [inline client/src/island/hexalot.ts:21]: Weird formatting

Comment (mitchelljustin) [reply client/src/island/hexalot.ts:21]: Suggest an alternative (ctrl-g) and I'll accept

File: server/index.ts (hunk 2 above)
Comment (mitchelljustin) [reply server/index.ts]: Right, but there is a hardcoded set of islands is there not?

Comment (geralddejong) [reply server/index.ts]: We could start with just one hardcoded, but eventually it'd be good to be able to choose which island to visit.

**Review by geralddejong — APPROVED** [review]

**Pull request merged**